### PR TITLE
native kernel for funcs: ones, ones_like, zeros, zeros_like, full, full_like

### PR DESCRIPTION
**Pull request by Alexander-Makaryev**

### CODE REVIEW

**Review by shssf — APPROVED** [review]

File: dpnp/dpnp_algo/dpnp_algo.pyx [hunk 1]
```diff
@@ -131,10 +133,19 @@ cpdef dparray dpnp_astype(dparray array1, dtype_target):
...
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(dtype)
+
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_INITVAL, param1_type, param1_type)
+
+    result_type = dpnp_DPNPFuncType_to_dtype( < size_t > kernel_data.return_type)
     cdef dparray result = dparray(shape, dtype=dtype)
 
-    for i in range(result.size):
-        result[i] = value
+    # TODO: find better way to pass single value with type conversion
```
Comment (shssf) [inline dpnp/dpnp_algo/dpnp_algo.pyx:143]: Absolutely agree, this is what we need to think about.

File: dpnp/backend/kernels/dpnp_krnl_common.cpp [hunk 2]
```diff
@@ -293,6 +217,111 @@ void dpnp_eigvals_c(const void* array_in, void* result1, size_t size)
...
+template <typename _DataType>
+void dpnp_initval_c(void* result1, void* value, size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+    _DataType val = *(reinterpret_cast<_DataType*>(value));
```
Comment (shssf) [inline dpnp/backend/kernels/dpnp_krnl_common.cpp:232]:
```suggestion
    _DataType *val = *(reinterpret_cast<_DataType*>(value));
```

**Comment by Alexander-Makaryev** [comment]
@shssf Tests are passed

**Review by PokhodenkoSA — APPROVED** [review]
LGTM

**Review by shssf — APPROVED** [review]

**Pull request merged**